src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java

```diff
@@ -829,6 +831,99 @@ public void testPrintTreeOption() throws Exception {
         Main.main("-t", getPath("InputMain.java"));
     }
 
+    @Test
+    public void testPrintXpathOption() throws Exception {
+        final String expected = addEndOfLine(
+            "CLASS_DEF -> CLASS_DEF [3:0]",
+            "`--OBJBLOCK -> OBJBLOCK [3:28]",
+            "    |--METHOD_DEF -> METHOD_DEF [4:4]",
+            "    |   `--SLIST -> { [4:20]",
+            "    |       |--VARIABLE_DEF -> VARIABLE_DEF [5:8]",
+            "    |       |   |--IDENT -> a [5:12]");
+        exit.checkAssertionAfterwards(() -> {
+            Assert.assertThat("Unexpected output log", systemOut.getLog(), is(expected));
+            Assert.assertThat("Unexpected system error log", systemErr.getLog(), is(""));
+        });
+        Main.main("-b", "/CLASS_DEF//METHOD_DEF[./IDENT[@text='methodOne']]//VARIABLE_DEF/IDENT",
+            getPath("InputMainXPath.java"));
+    }
+
+    @Test
+    public void testPrintXpathCommentNode() throws Exception {
+        final String expected = addEndOfLine(
+            "CLASS_DEF -> CLASS_DEF [17:0]",
+            "`--OBJBLOCK -> OBJBLOCK [17:19]",
+            "    |--CTOR_DEF -> CTOR_DEF [19:4]",
+            "    |   |--BLOCK_COMMENT_BEGIN -> /* [18:4]");
+        exit.checkAssertionAfterwards(() -> {
+            Assert.assertThat("Unexpected output log", systemOut.getLog(), is(expected));
+            Assert.assertThat("Unexpected system error log", systemErr.getLog(), is(""));
+        });
+        Main.main("-b", "/CLASS_DEF//BLOCK_COMMENT_BEGIN",
+            getPath("InputMainXPath.java"));
+    }
+
+    @Test
+    public void testPrintXpathNodeParentNull() throws Exception {
+        final String expected = "PACKAGE_DEF -> package [1:0]" + EOL;
+        exit.checkAssertionAfterwards(() -> {
+            Assert.assertThat("Unexpected output log", systemOut.getLog(), is(expected));
+            Assert.assertThat("Unexpected system error log", systemErr.getLog(), is(""));
+        });
+        Main.main("-b", "/PACKAGE_DEF", getPath("InputMainXPath.java"));
+    }
+
+    @Test
+    public void testPrintXpathFullOption() throws Exception {
+        final String expected = addEndOfLine(
+            "CLASS_DEF -> CLASS_DEF [3:0]",
+            "`--OBJBLOCK -> OBJBLOCK [3:28]",
+            "    |--METHOD_DEF -> METHOD_DEF [8:4]",
+            "    |   `--SLIST -> { [8:26]",
+            "    |       |--VARIABLE_DEF -> VARIABLE_DEF [9:8]",
+            "    |       |   |--IDENT -> a [9:12]");
+        exit.checkAssertionAfterwards(() -> {
+            Assert.assertThat("Unexpected output log", systemOut.getLog(), is(expected));
+            Assert.assertThat("Unexpected system error log", systemErr.getLog(), is(""));
+        });
+        Main.main("--tree-branch-for-xpath", "/CLASS_DEF//METHOD_DEF[./IDENT[@text='method']]"
```

Comment:
```
    |--METHOD_DEF -> METHOD_DEF [8:4]
    |   |--MODIFIERS -> MODIFIERS [8:4]
    |   |--TYPE -> TYPE [8:4]
    |   |   `--LITERAL_VOID -> void [8:4]
    |   |--IDENT -> method [8:9]
    |   |--LPAREN -> ( [8:15]
    |   |--PARAMETERS -> PARAMETERS [8:16]
    |   |   `--PARAMETER_DEF -> PARAMETER_DEF [8:16]
    |   |       |--MODIFIERS -> MODIFIERS [8:16]
    |   |       |--TYPE -> TYPE [8:16]
    |   |       |   `--LITERAL_INT -> int [8:16]
    |   |       `--IDENT -> param [8:20]
    |   |--RPAREN -> ) [8:25]
    |   `--SLIST -> { [8:26]
    |       |--VARIABLE_DEF -> VARIABLE_DEF [9:8]
    |       |   |--MODIFIERS -> MODIFIERS [9:8]
    |       |   |--TYPE -> TYPE [9:8]
    |       |   |   `--LITERAL_INT -> int [9:8]
    |       |   |--IDENT -> a [9:12]
    |       |   `--ASSIGN -> = [9:14]
    |       |       `--EXPR -> EXPR [9:16]
    |       |           `--NUM_INT -> 10 [9:16]
    |       |--SEMI -> ; [9:18]
    |       `--RCURLY -> } [10:4]
```

I am not sure why ` |--IDENT -> a [9:12]` is selected by `IDENT[@text='method']`, I think it should be `IDENT -> method [8:9]`

Reply:
Sorry, didn't understand.
Current xpath is for variable name in method "method". Same variable is in another method so we use qualifier to distinguish method we need
![image](https://user-images.githubusercontent.com/8901354/64318386-97dda080-cfc2-11e9-80f0-f4a4a6376899.png)

Reply:
Now I see a problem, I didn't noticed second line of xpath. 
Is it possible to move definition of xpath to special variable ? To keep whole xpath as single string.

Reply:
Ok, I corrected this